source/class/qx/ui/core/LayoutItem.js

```diff
@@ -792,6 +792,14 @@ qx.Class.define("qx.ui.core.LayoutItem",
      */
     setUserBounds : function(left, top, width, height)
     {
+      
+      if (!this.__userBounds) {
+        var parent = this.$$parent;
+        if (parent) {
+          parent.updateLayoutProperties();
+        }
```

Comment:
`__userBounds` has not yet been set (it happens later, at line 803). Will there be any effect of calling `parent.updateLayoutProperties()` when there were no user bounds and there are still no user bounds? Won't that leave the layout exactly as it was? Should this be called after setting `__userBounds`?

Reply:
I have a similar question ... if userBounds are already set, and get changed, then the updateLayoutParent would not be called ?

Reply:
The issue is different:
- create a widget without user bounds and add it to a container with Dock or Table layouts (the two layouts that cache the children list using getLayoutChildre(), which will include the widget just added)
- call setUserBounds() on the widget, it will move the widget but as soon as the parent performs a layout again it will break the layout and treat the widget as a docking child since the layout children list has not been updated.
- same thing happens the other way around, resetting the user bounds will not update the layout children list in the parent, unless something else causes the _invalidChildrenCache flag to be set.

Calling LayoutItem.updateLayoutProperties() is just to keep the same coding style, otherwise the call should be parent.invalidateLayoutChildren() which end up in the same place: just setting layout._invalidChildrenCache to true. So at the next layout round it will discard the cached layout children list and rebuild it with or without the user bounds children.